### PR TITLE
Optimize marshalling for jsonb types

### PR DESCRIPTION
**Pull request by agnivade**
We check for the presence of binary_parameters
in the DSN and add the 0x01 byte accordingly.

This helps us avoid casting to string
and efficiently use the database.

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/store.go [hunk 1]
```diff
@@ -323,6 +324,25 @@ func (ss *SqlStore) DriverName() string {
...
 }
 
+// IsBinaryParamEnabled returns whether the data source uses binary_parameters
+// when using Postgres
+func (ss *SqlStore) IsBinaryParamEnabled() (bool, error) {
+	if ss.DriverName() != model.DatabaseDriverPostgres {
+		return false, nil
+	}
+
+	url, err := url.Parse(*ss.settings.DataSource)
```
Comment (isacikgoz) [inline store/sqlstore/store.go:334]: Should we compute this every time? Maybe we could just set a private boolean field?

**Review by isacikgoz — COMMENTED** [review]
Looks good, just a question below 👇

File: store/sqlstore/store.go (hunk 1 above)
Comment (agnivade) [reply store/sqlstore/store.go:334]: Excellent idea

**Review by isacikgoz — APPROVED** [review]
Awesome!

**Review by koox00 — APPROVED** [review]

**Comment by agnivade** [comment]
@saturninoabril - Whenever you get a chance :)

**Review by saturninoabril — APPROVED** [review]
Thanks @agnivade! All good.

**Pull request merged**